tests/post/features/ingress.feature

```diff
@@ -19,12 +19,36 @@ Feature: Ingress
         When we perform an HTTP request on port 80 on a control-plane IP
         Then the server should not respond
 
+    Scenario: Failover of Control Plane Ingress VIP using MetalLB
+        Given the Kubernetes API is available
+        And we are on a multi node cluster
+        And MetalLB is enabled
+        When we stop the node hosting the Control Plane Ingress VIP
+        Then the node hosting the Control Plane Ingress VIP changed
+        And we are able to login to Dex as 'admin@metalk8s.invalid' using password 'password'
+
     Scenario: Change Control Plane Ingress IP to node-1 IP
         Given the Kubernetes API is available
         And we are on a multi node cluster
         And pods with label 'app.kubernetes.io/name=ingress-nginx' are 'Ready'
-        When we update control plane ingress IP to node 'node-1' IP
+        When we disable MetalLB and set control plane ingress IP to node 'node-1' IP
         And we wait for the rollout of 'daemonset/ingress-nginx-control-plane-controller' in namespace 'metalk8s-ingress' to complete
         And we wait for the rollout of 'deploy/dex' in namespace 'metalk8s-auth' to complete
         Then the control plane ingress IP is equal to node 'node-1' IP
         And we are able to login to Dex as 'admin@metalk8s.invalid' using password 'password'
+
+    Scenario: Change Control Plane Ingress IP to a new VIP
+        Given the Kubernetes API is available
+        And MetalLB is enabled
+        And pods with label 'app.kubernetes.io/name=ingress-nginx' are 'Ready'
+        # In our CI we need a private network to use this "VIP" feature,
+        # assume when Control Plane network is equal to "192.168.1.0/24" that we are
+        # on a private network
+        And the control plane cidr is equal to '192.168.1.0/24'
```

Comment:
Mmh, not super fond of this approach, where the test scenario depends on the execution context :confused: 

Could we instead define:

- a command-line option for pytest to pass a VIP for control plane
- a `given` saying "a VIP for control plane is available", which checks if the VIP correctly belongs to the control plane CIDR (if needed, not sure), and stores its value in a fixture
- a `when` saying "we set control plane ingress IP to the available VIP", which uses the aforementioned fixture

?

Reply:
> VIP correctly belongs to the control plane CIDR (if needed, not sure)

It's not strictly needed we just need the IP to be routed on the Control Plane nodes

But right I can likely make something better with a variable but it means we need to pass this variable to run this test where, with the current approach, we always run this test if we use MetalLB but ... ok I will do with a variable it cleaner